query-graphs-js/src/graphPath.ts

```diff
@@ -824,12 +824,18 @@ export function advancePathWithTransition<V extends Vertex>(
       if (type && isCompositeType(type) && type.field(fieldName)) {
         // That subgraph has the type we look for, but we have recorded no dead-ends. This means there is no edge to that type,
         // and thus that it has no keys.
-        assert(!isEntityType(type), () => `Expected type ${type} in ${subgraph} to not have keys`);
+        const metadata = federationMetadata(schema);
+        const keys: Directive<CompositeType, {fields: any, resolvable?: boolean}>[] = metadata ? type.appliedDirectivesOf(metadata.keyDirective()) : [];
```

Comment:
no need to explicitly declare the type here since we don't want to have to change in multiple places if we add a new field down the line.